app/services/marketplace_service/transaction.rb

```diff
@@ -253,13 +253,20 @@ def transaction(transaction_id)
           .or_else(nil)
       end
 
-      def transaction_with_conversation(transaction_id, person_id, community_id)
-        Maybe(TransactionModel.joins(:listing)
+      def transaction_with_conversation(transaction_id:, person_id: nil, community_id:)
+        rel = TransactionModel.joins(:listing)
           .where(id: transaction_id)
           .where(community_id: community_id)
           .includes(:booking)
-          .where("starter_id = ? OR listings.author_id = ?", person_id, person_id)
-          .first)
+
+        with_person =
+          if person_id
+            rel.where("starter_id = ? OR listings.author_id = ?", person_id, person_id)
+          else
+            rel
+          end
```

Comment:
Actually, this didn't quite work. 

The fact that the `rel` is a magical ActiveRecord relationship made this a bit tricky. After mapping, Maybe tries to define whether the value is a non-value (None) or a value (Some). The way it tests it is to see if the value is null, or if the length of the value is 0. Asking the length from the `rel` runs the query, and that's not actually what we wanted. Running the query returns an empty array and that maps to None, and then the `or_else` is run.

The solution was to wrap the `rel` in another array.